JavaFiddle-ejb/src/main/java/ru/ncedu/dynamic/DynamicCompiler.java

```diff
@@ -18,59 +19,63 @@
-    }
+//    public static final String JUNIT_LOCATION;
+//    public static final String FRWK_LOCATION;
+//    public static final String DEFAULT_SERVER_LOG;
+
+//    static {
+//        Properties compilerProps = new Properties();
+//        try {
+//            compilerProps.load(DynamicCompiler.class.getClassLoader().getResourceAsStream("ru/ncedu/test/JavaBenchProperties.properties"));
+//        } catch (IOException ex) {
+//            LOG.error("", ex);
+//        }
+//
+//        JUNIT_LOCATION = compilerProps.getProperty("JUNIT_LOCATION");
+//        FRWK_LOCATION = compilerProps.getProperty("FRWK_LOCATION");
+//        DEFAULT_SERVER_LOG = compilerProps.getProperty("DEFAULT_SERVER_LOG");
+//        LOG.trace("JUNIT_LOCATION=" + JUNIT_LOCATION);
+//        LOG.trace("FRWK_LOCATION=" + FRWK_LOCATION);
+//        LOG.trace("DEFAULT_SERVER_LOG=" + DEFAULT_SERVER_LOG);  // We have to change server log in JavaBenchProperties
+//    }
 
     public void init(ClassLoader parentLoader) throws Exception {
 
         compiler = ToolProvider.getSystemJavaCompiler();
-
+        BasicConfigurator.configure();
         collector = new DiagnosticCollector<>();
         StandardJavaFileManager fman = compiler.getStandardFileManager(null, null, null);
-        fman.setLocation(StandardLocation.CLASS_PATH, Arrays.asList(new File(
-                JUNIT_LOCATION),
-                new File(FRWK_LOCATION)
-        ));
+//        fman.setLocation(StandardLocation.CLASS_PATH, Arrays.asList(new File(
+//                JUNIT_LOCATION),
+//                new File(FRWK_LOCATION)
+//        ));
         manager = new DynamicClassFileManager(fman, parentLoader);
 
     }
 
-    public String compileToClass(List<SimpleJavaFileObject> sources)
+    protected String compileToClass(List<SimpleJavaFileObject> sources)
             throws Exception {
 
         long nanos = System.nanoTime();
 
         ArrayList<SimpleJavaFileObject> sourceFiles = new ArrayList<>();
-        ArrayList<SimpleJavaFileObject> otherFiles = new ArrayList<>();       
+        ArrayList<SimpleJavaFileObject> otherFiles = new ArrayList<>();
         for (SimpleJavaFileObject source : sources) {
-            
+
             if (source instanceof StringJavaFileObject) {
                 sourceFiles.add(source);
             } else if (source instanceof ByteArrayResource) {
                 otherFiles.add(source);
             }
-            
+
         }
         ArrayList<String> javaOptions = new ArrayList<>();
-        // javaOptions.add("-g");
-        // javaOptions.add("-verbose");
+//         javaOptions.add("-g");
+//         javaOptions.add("-verbose");
+
+//        javaOptions.add("-bootclasspath");
+//        javaOptions.add("C:\\Program Files\\Java\\jdk1.7.0_79\\jre\\lib\\rt.jar");
+
         Iterable<? extends JavaFileObject> units = sourceFiles;
         CompilationTask task = compiler.getTask(null, manager, collector, javaOptions,
```

Comment:
Did you try passing custom Writer instance to the first parameter of compiler.getTask() to be used instead of System.err as stated in JavaDocs: http://docs.oracle.com/javase/7/docs/api/javax/tools/JavaCompiler.html
Will its output be sufficient to remove hacks with System.out?